packages/core/src/Stores/notification-store.js

```diff
@@ -1640,4 +1652,50 @@ export default class NotificationStore extends BaseStore {
             platform: 'Account',
         });
     };
+
+    handleCurrencyRemovalNotification = (account_currency_closure_status, currency) => {
+        const is_funded_account = account_currency_closure_status.status_codes.includes('funded_account');
+        const is_non_funded_account = account_currency_closure_status.status_codes.includes('non_funded_account');
+
+        if (!is_funded_account && !is_non_funded_account) return;
+
+        const notification_header = is_funded_account
+            ? localize('Withdraw your funds')
+            : localize('Change your currency');
+
+        const time_of_closure = getDateFromTimestamp(account_currency_closure_status.time_of_closure);
+
+        const notification_message = is_funded_account ? (
+            <Localize
+                i18n_default_text="{{currency}} accounts won't be available after {{time_of_closure}}."
+                values={{ time_of_closure, currency }}
+            />
+        ) : (
+            <Localize
+                i18n_default_text="{{currency}} accounts won't be available after {{time_of_closure}}. Choose a new account currency."
+                values={{ time_of_closure, currency }}
+            />
+        );
+
+        const notification_button_action = is_funded_account
+            ? {
+                  route: routes.cashier_withdrawal,
+                  text: <Localize i18n_default_text='Withdraw {{currency}}' values={{ currency }} />,
+              }
+            : {
+                  text: localize('Contact live chat'),
+                  onClick: () => {
+                      window.LC_API.open_chat_window();
+                  },
+              };
+
+        this.addNotificationMessage({
```

Comment:
why not we have 2 objects -  funded_account and one for non-funded account. In this way we can reduce checking is_funded_account for every field. something like:
```

   key: 'funded_account_currency_closure',
            header: localize('Withdraw your funds'),
            message:  <Localize
                i18n_default_text="{{currency}} accounts won't be available after {{time_of_closure}}."
                values={{ time_of_closure, currency }}
            />
            action: {
                  route: routes.cashier_withdrawal,
                  text: <Localize i18n_default_text='Withdraw {{currency}}' values={{ currency }} />,
              }
            type: 'warning',
            should_show_again: true,
        });
```

Reply:
I think it is almost the same. 
having two objects just some lines of code will be duplicated, and we need to maintain two notifications keys instead of one.

Reply:
yea.. just that other was more readable to me